Fabric.Identity.API/scripts/tests/Install-Identity-Utilities.tests.ps1

```diff
@@ -0,0 +1,761 @@
+                # Assert
+                $appEndpoint | Should -Be "https://host.fabric.local/identity"
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Get-FullyQualifiedMachineName -Times 1 -Exactly
+            }
+        }
+    }
+}
+
+Describe 'Test-ShouldShowCertMenu'{
+    InModuleScope Install-Identity-Utilities{
+        Context 'Interactive Mode'{
+            It 'Should return true when in interactive mode and signing cert thumprint is not present'{
+                $shouldShow = Test-ShouldShowCertMenu -primarySigningCertificateThumbprint "" -encryptionCertificateThumbprint "12345" -quiet $false
+                $shouldShow | Should -Be $true
+            }
+            It 'Should return true when in interactive mode and encryption cert thumprint is not present'{
+                $shouldShow = Test-ShouldShowCertMenu -primarySigningCertificateThumbprint "12345" -encryptionCertificateThumbprint "" -quiet $false
+                $shouldShow | Should -Be $true
+            }
+            It 'Should return false when in interactive mode and encryption and signing cert thumbprint are present'{
+                $shouldShow = Test-ShouldShowCertMenu -primarySigningCertificateThumbprint "12345" -encryptionCertificateThumbprint "12345" -quiet $false
+                $shouldShow | Should -Be $false
+            }
+            It 'Should return true when in interactive mode and no thumbprints are present'{
+                $shouldShow = Test-ShouldShowCertMenu -primarySigningCertificateThumbprint "" -encryptionCertificateThumbprint "" -quiet $false
+                $shouldShow | Should -Be $true
+            }
+        }
+        Context 'Quiet Mode'{
+            It 'Should return false when in quiet mode and signing and encryption cert thumbprnts are present'{
+                $shouldShow = Test-ShouldShowCertMenu -primarySigningCertificateThumbprint "12345" -encryptionCertificateThumbprint "12345" -quiet $true
+                $shouldShow | Should -Be $false
+            }
+            It 'Should return false when in quiet mode and signing cert thumbprint is not present'{
+                $shouldShow = Test-ShouldShowCertMenu -primarySigningCertificateThumbprint "" -encryptionCertificateThumbprint "12345" -quiet $true
+                $shouldShow | Should -Be $false
+            }
+            It 'Should return false when in quiet mode and encryption cert thumbprint is not present'{
+                $shouldShow = Test-ShouldShowCertMenu -primarySigningCertificateThumbprint "12345" -encryptionCertificateThumbprint "" -quiet $true
+                $shouldShow | Should -Be $false
+            }
+            It 'Should return false when in quiet mode and no thumbprints are present'{
+                $shouldShow = Test-ShouldShowCertMenu -primarySigningCertificateThumbprint "" -encryptionCertificateThumbprint "" -quiet $true
+                $shouldShow | Should -Be $false
+            }
+        }
+    }
+}
```

Comment:
We discussed writing tests for the installer at some point in the past, it's nice finally having some written though :)